### PR TITLE
Added a shopping cart to the UserModel constructor that is used for user registeration.

### PR DESCRIPTION
**Pull request by khalidkanaan**

### CODE REVIEW

File: src/main/java/main/controller/CartController.java [hunk 1]
```diff
@@ -27,8 +27,8 @@ public CartController(UserService userService) {
 
 
     @GetMapping(value="userCart", produces= MediaType.APPLICATION_JSON_VALUE)
-    public Map<BookModel, Integer> getBooks(@RequestParam("userid") long userid) {
-        UserModel user = userRepository.findById(userid);
```
Comment (khalidkanaan) [inline src/main/java/main/controller/CartController.java:31]: For some reason findById here results in a build error so I opted to use findAllById instead. It's not pretty but it works.

Build error:
JpaRepository issue. java: cannot access java.util.Optional class file for java.util.Optional not found

**Comment by khalidkanaan** [comment]
image

File: src/main/java/main/controller/CartController.java (hunk 1 above)
Comment (Mohman556) [reply src/main/java/main/controller/CartController.java:31]: If it works, it works! :)

**Review by khalidkanaan — COMMENTED** [review]
For some reason the ID of gets incremented too in the database, I don't know why this happens but it shouldn't cause any issues.
image

**Comment by Mohman556** [comment]
> For some reason the ID of gets incremented too in the database, I don't know why this happens but it shouldn't cause any issues. image

We should investigate this just in this case, but I agree, I don't think it should an issue.

**Review by Mohman556 — APPROVED** [review]
Nice fixes

**Pull request merged**